### PR TITLE
support for views-jte

### PR DESCRIPTION
**Pull request by edward3h**
I added [JTE support in micronaut-views](https://github.com/micronaut-projects/micronaut-views/pull/320).
This adds it to Launch.

### CODE REVIEW

File: starter-core/src/main/java/io/micronaut/starter/feature/view/JTE.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                .extension(new RockerWritable(mvnPluginJTE.template(coordinate.getGroupId(),
+                        coordinate.getArtifactId(),
+                        coordinate.getVersion(),
+                        jteSrcDir)))
+                .build());
+        generatorContext.addTemplate("exampleJte", new RockerTemplate(jteSrcDir + "/example.jte", exampleJTE.template()));
+    }
+
+    private String jteSrcDir(GeneratorContext generatorContext) {
+        String path = generatorContext.getConfiguration().getPath();
```
Comment (jameskleeh) [inline starter-core/src/main/java/io/micronaut/starter/feature/view/JTE.java]: You shouldn't be doing path calculation, just specify the full path.

`src/main/resources/jte/example.jte` for example

File: starter-core/src/main/resources/pom.xml [hunk 2]
```diff
@@ -157,5 +157,21 @@
             <artifactId>plugin</artifactId>
             <version>5.12.0</version>
         </dependency>
+        <dependency>
+            <!-- FIXME: this can be removed once the micronaut-bom references version 3.1.2 or higher -->
+            <groupId>io.micronaut.views</groupId>
+            <artifactId>micronaut-views-jte</artifactId>
+            <version>3.1.2</version>
```
Comment (jameskleeh) [inline starter-core/src/main/resources/pom.xml]: This can be removed

File: starter-core/src/main/java/io/micronaut/starter/feature/view/JTE.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    @Override
+    public String getMicronautDocumentation() {
+        return "https://micronaut-projects.github.io/micronaut-views/latest/guide/#jte";
+    }
+
+    @Override
+    public void apply(GeneratorContext generatorContext) {
+        generatorContext.addDependency(Dependency.builder()
+                .groupId("io.micronaut.views")
+                .lookupArtifactId("micronaut-views-jte")
```
Comment (jameskleeh) [inline starter-core/src/main/java/io/micronaut/starter/feature/view/JTE.java]: This should just be artifactId

**Review by jameskleeh — CHANGES_REQUESTED** [review]

File: starter-core/src/main/java/io/micronaut/starter/feature/view/JTE.java (hunk 3 above)
Comment (edward3h) [reply starter-core/src/main/java/io/micronaut/starter/feature/view/JTE.java]: changed

**Comment by edward3h** [comment]
I have some failing tests now - will fix.

File: starter-core/src/main/resources/pom.xml (hunk 2 above)
Comment (edward3h) [reply starter-core/src/main/resources/pom.xml]: My build is currently pulling micronaut-bom-3.2.8-SNAPSHOT and this still refers to micronaut-views-bom:3.0.4, which doesn't include micronaut-views-jte.

Is there something else I need to do to pick up the newer version of micronaut-views?

**Comment by sdelamo** [comment]
Closed via https://github.com/micronaut-projects/micronaut-starter/pull/1045

**Pull request closed without merging**